soroban-auth/src/lib.rs

```diff
@@ -11,6 +11,7 @@ pub use crate::public_types::{
 fn check_ed25519_auth(env: &Env, auth: &Ed25519Signature, function: Symbol, args: Vec<RawVal>) {
     let msg = SignaturePayloadV0 {
         function,
+        call_cntxt: env.get_parent_call_stack(),
```

Comment:
I don't think we can just use the call stack here. Consider contract A that calls contract B. We want to authorize B to do something (e.g. transfer token). Then we call A with `(A_signature, B_signature, A_args)`, but then `B_signature` will be in the call stack and we won't be able to sign this call stack in the first place.

Reply:
The call stack doesn't include the parameters for the reason you mentioned. It's just a `Vec<(BytesN<32>/*contract ID*/, Symbol/*function name*/)>`.

Reply:
Oh, I see. It's a bit counterintuitive, but it seems to me that as long as we check the identity of the parent contract caller to be the same as the identity of the sub-contract signer, the args are not necessary to be signed. It also seems like including args opens up some weird possibilities that are probably not that useful (though I'm not sure about that).